service/chartconfig/v1/resource/chart/update.go

```diff
@@ -2,14 +2,81 @@ package chart
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/key"
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	chartState, err := toChartState(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating chart %s", chartState.ChartName))
+
+	name := key.ChartName(customObject)
+	releaseName := chartState.ReleaseName
+	channel := chartState.ChannelName
+
+	tarballPath, err := r.apprClient.PullChartTarball(name, channel)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.helmClient.UpdateReleaseFromTarball(releaseName, tarballPath)
```

Comment:
What if newUpdateChange returns empty ChartState?

Reply:
Agreed if the ChartState is empty no action should be taken.

Reply:
Done, currently `toChartState` returns an empty `ChartState` on nil input, i've changed `ApplyUpdateChange` to check for an empty `ReleaseName` field, PTAL.

Reply:
Thanks LGTM.